source/OptimaJet.DWKit.StarterApplication/Repositories/ProjectRepository.cs

```diff
@@ -16,67 +17,58 @@ public class ProjectRepository : ControllerRepository<Project>
         public IOrganizationContext OrganizationContext { get; }
         public ICurrentUserContext CurrentUserContext { get; }
         public UserRepository UserRepository { get; set; }
+        public GroupRepository GroupRepository { get; set; }
+        public IEntityRepository<Organization> OrganizationRepository { get; set; }
 
         public ProjectRepository(
             ILoggerFactory loggerFactory,
             IJsonApiContext jsonApiContext,
             IOrganizationContext organizationContext,
             ICurrentUserContext currentUserContext,
             UserRepository userRepository,
+            GroupRepository groupRepository,
+            IEntityRepository<Organization> organizationRepository,
             IDbContextResolver contextResolver
             ) : base(loggerFactory, jsonApiContext, contextResolver)
         {
             this.OrganizationContext = organizationContext;
             this.CurrentUserContext = currentUserContext;
             this.UserRepository = userRepository;
+            this.GroupRepository = groupRepository;
+            this.OrganizationRepository = organizationRepository;
         }
-
-        public override IQueryable<Project> Get()
+        public override IQueryable<Project> Filter(IQueryable<Project> query, FilterQuery filterQuery)
```

Comment:
I worked with this for quite a while.  A few issues like the three GetWithOrganizationHeaders are not quite sufficiently alike to just have one.  The user one is somewhat different than the other two.  The final return filterQuery is also a problem since what it really returns at that point is a call to the base.Filter method which would be a bit hard to call from here I think.  You could maybe pass another Func that would point.  Some other things.  Finally decided after working on it for quite a while that I needed to get the other issues resolved and this one really is not creating any performance issues as is.